app/services/page_updater.rb

```diff
@@ -0,0 +1,63 @@
+class PageUpdater
+  include Ak::Client
+
+  attr_accessor :params
+
+  def self.run(params)
+    new(params).run
+  end
+
+  def initialize(params)
+    @params = params
+  end
+
+  def run
+    data = params[:params]
+    data.delete(:name)
+    @campaign_id = data.delete(:campaign_id)
+
+    raise ArgumentError, "Missing resource URI for page" if params[:petition_uri].blank? and params[:donation_uri].blank?
+
+    unless params[:petition_uri].blank?
```

Comment:
@osahyoun is not exactly the same behaviour. The if above checks if both `petition_url` and `donation_url` are blank, below we're checking for each one separately.

And btw, I just copy/pasted this code from the controller, I didn't get to refactor it.